src/jit/valuenum.cpp

```diff
@@ -1613,6 +1613,7 @@ double ValueNumStore::GetConstantDouble(ValueNum argVN)
 {
     assert(IsVNConstant(argVN));
     var_types argVNtyp = TypeOfVN(argVN);
+    assert(varTypeIsFloating(argVNtyp));
```

Comment:
`ValueNumStore::GetConstantDouble` currently returns for both `float` and `double` constants. This means in some cases we are upcasting, executing some code, and then downcasting back to `float`.

After the reworking of `EvalMathFuncUnary` there are still a couple of places where this is happening (a simple search for `GetConstantDouble` in this file will show them). These calls could probably be rewritten to no longer up/down cast, but I would like some input before I do that work.